src/modules/ui-actions.js

```diff
@@ -18,9 +16,16 @@ const addItem = (todos, input, render) => {
   render();
 };
 
-const clearCompleted = (todos, render) => {
-  todos = todos.filter((todo) => !todo.completed);
-  render();
+const clearCompleted = (todos) => {
+  const updatedTodos = todos.filter((todo) => !todo.completed);
+
+  // Update the indexes of remaining todos
+  updatedTodos.forEach((todo, index) => {
+    todo.index = index;
+  });
+
+  localStorage.setItem('todos', JSON.stringify(updatedTodos));
+  return updatedTodos;
 };
```

Comment:
## Optional Changes 
- [ ] This feature is not part of the project requirement, at least not yet. It's meant for future implementation and having it makes your app almost looks complete. This will make your project seen as a plagiarized project. It would be nice if you could remove the clearCompleted feature and do only what you require as features are built on top of each other step by step.